uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js

```diff
@@ -55,24 +57,35 @@ define([
           public_holiday: isPublicHoliday || false
         };
 
-        deferred.resolve(this.sendGET('LeaveRequest', 'getbalancechangebyabsencetype', params)
+        return this.sendGET('LeaveRequest', 'getbalancechangebyabsencetype', params)
           .then(function (data) {
             return data.values;
-          }));
-
-        return deferred.promise;
+          });
       },
 
       /**
        * This method is used to update a leave request
        *
-       * @param {object} updatedAttributes - Updated values of leave request
+       * @param {object} params - Updated values of leave request
        * @return {Promise} Resolved with {Object} Updated Leave request
        */
-      update: function (updatedAttributes) {
+      update: function (params) {
         $log.debug('LeaveRequestAPI.update');
+        var deferred = $q.defer();
 
-        return this.sendPOST('LeaveRequest', 'create', updatedAttributes);
+        if (!('id' in params)) {
+          deferred.reject({
+            is_error: 1,
+            error_message: 'id is mandatory field'
+          });
+          return deferred.promise;
+        }
+
+        return this.sendPOST('LeaveRequest', 'create', params)
+          .then(function (data) {
+            //returns array of single object hence getting first object
```

Comment:
sequential 0 will return 
thank you for the tip,
using sequential 0 will return object indexed by id, so its not very useful
```json
{

    "is_error": 0,
    "version": 3,
    "count": 1,
    "id": 24,
    "values": {
        "24": {
            "id": "24",
            "type_id": "3",
            "contact_id": "202",
            "status_id": "3",
            "from_date": "20161213000000",
            "from_date_type": "",
            "to_date": "",
            "to_date_type": ""
        }
    }
}
```